### ISSUE TITLE
Add live region to announce when Select opens

### ISSUE THREAD

**Issue by IanVS** [ariakit/ariakit#2510]
### Motivation

Downshift automatically adds a [status div](https://github.com/downshift-js/downshift/blob/89395a5691ccd96836e27426d66f92fc894a646b/src/set-a11y-status.js#L26) to [announce](https://github.com/downshift-js/downshift/blob/89395a5691ccd96836e27426d66f92fc894a646b/src/utils.js#LL150C10-L150C30) status changes to Selects and Comboboxes.  From what I can tell, ariakit does not do this, but it would be a nice feature for those using assistive technology.  

I'd particularly like to use this in an async Select component, to indicate to the user that results are loading or if there has been an error.

### Usage example

I think as a start, the status could be built in and not overridable.  Or, if it should be, maybe it could be done in a prop to `useSelectStore`, like so:

```ts
const select = useSelectStore({a11yMessage: ({isOpen, resultCount, previousResultCount}) => 'message'})
```

### Requirements

- The status should announce when the select box is opened
- It should be visually hidden
- It should re-evaluate when the open state or count of available options changes

### Workaround

This can likely be created by the user, but I haven't attempted it so far.  If I do, I'll post my solution back here.

**Comment by diegohaz** [comment]
I have a rough idea for a general purpose Notification component that could be used for toasts and visually hidden announcements. Here's a gist of my initial idea for the API: https://gist.github.com/diegohaz/341ed774ebac3b4dd820ae48d8bc822c

Also related to #464

**Comment by IanVS** [comment]
Is this still something you're considering?

**Comment by diegohaz** [comment]
A Notification module? Yes, it's still in the plans.

**Comment by IanVS** [comment]
Thanks, I saw this became stale and the referenced issue is closed as completed.

**Comment by IanVS** [comment]
Not stale?
